src/ServicePulse.Host/app/js/directives/ui.particular.largeGraph.js

```diff
@@ -0,0 +1,113 @@
+﻿(function(window, angular, undefined) {
+    'use strict';
+
+    angular.module('ui.particular.largeGraph', [])
+        .directive('largeGraph',
+            function() {
+                return {
+                    restrict: 'E',
+                    scope: {
+                        data: '=plotPoints',
+                        avg: '=plotAverage',
+                        dates: '=xaxisPoints',
+                        width: '=plotWidth',
+                        heigth: '=plotHeight'
+                    },
+                    template: '<svg></svg>',
+                    link: function link(scope, element, attrs) {
+                        scope.$watch('data', function () {
+                            d3.selectAll("svg > *").remove();
```

Comment:
If the watch goes then this remove can also go

Reply:
And as I refresh the control old rendered values stayed.